src/xo-app/host/tab-network.js

```diff
@@ -91,7 +91,7 @@ class ConfigureIpModal extends Component {
 }))
 class PifItem extends Component {
   componentWillMount () {
-    getIpv4ConfigModes().then(configModes =>
+    getIpv4ConfigModes().then((configModes = []) =>
```

Comment:
Also, you probably need to initialize the state, because the component may render before `getIpv4ConfigModes()` as returned.